### DISCUSSION TITLE
Add enums & types for HTTP code & reason phrases.

### DISCUSSION THREAD

**Discussion by VaibhavAcharya** [vercel/next.js#27586] [Ideas]
### Describe the feature you'd like to request

Currently the `status` function of `NextApiResponse` type accepts any `number`.
It is difficult to find right status code every time you work with API's.
Also it is very possible to set a wrong HTTP code due to no type checking available.

### Describe the solution you'd like

Provide an object for all status code and their reason phrases and apply correct strict type checks everywhere they are required.
This would certainly make DX of working with API's more smooth.

### Describe alternatives you've considered

Currently I am using [`http-status-codes`](https://github.com/prettymuchbryce/http-status-codes). It provides constants and types enumerating all HTTP status codes & reason phrases.
I think this should be by default available in `Next.js`.

**Comment by 13RTK** [comment]
I agree with you, but there is still no plan to integrate this feature😞
